### DISCUSSION TITLE
Info how to achieve some effects

### DISCUSSION THREAD

**Discussion by UkooLabs** [SixLabors/ImageSharp#1507] [Q&A]
Thought id reach out as far as I know ImageSharp does not have the following effects...

1) Enhance Shadow, Warmth, Highlights
2) Auto Color (Brightness / Contrast)
3) Color Tint (Not same as hue)

Does anyone have any info on how these may be achieved using ImageSharp?

**Comment by tocsoft** [comment]
## RE:Color Tint 

Source image:
![Orig](https://user-images.githubusercontent.com/166440/104643636-9b983180-56a4-11eb-9bc9-14023cc0a0b0.jpg)
BlueViolet
```csharp
using var bv = img.Clone(x => x
.SetGraphicsOptions(o => o.ColorBlendingMode = SixLabors.ImageSharp.PixelFormats.PixelColorBlendingMode.Multiply)
.Fill(Color.BlueViolet));
bv.Save("BlueViolet.jpg");
```
![BlueViolet](https://user-images.githubusercontent.com/166440/104643633-9aff9b00-56a4-11eb-90f5-baef7fc1e010.jpg)
Red
```csharp
using var bv = img.Clone(x => x
.SetGraphicsOptions(o => o.ColorBlendingMode = SixLabors.ImageSharp.PixelFormats.PixelColorBlendingMode.Multiply)
.Fill(Color.Red));
bv.Save("Red.jpg");
```
![Red](https://user-images.githubusercontent.com/166440/104643637-9b983180-56a4-11eb-91f8-415f87f4a133.jpg)
Salmon
```csharp
using var bv = img.Clone(x => x
.SetGraphicsOptions(o => o.ColorBlendingMode = SixLabors.ImageSharp.PixelFormats.PixelColorBlendingMode.Multiply)
.Fill(Color.Salmon));
bv.Save("Salmon.jpg");
```
![Salmon](https://user-images.githubusercontent.com/166440/104643639-9c30c800-56a4-11eb-9aa9-972f3b9088ef.jpg)

**Comment by UkooLabs** [comment]
Here is what i came up with that maybe useful for someone...    

        private float Lerp(float from, float to, float step)
        {
            return from + (to - from) * step;
        }

        private Color InterpolateColor(Color from, Color to, int steps, int step)
        {
            var fromRgba = from.ToPixel<Rgba32>();
            var toRgba = to.ToPixel<Rgba32>();
            float percent = step / (float)steps;
            var intervalR = Lerp(fromRgba.R, toRgba.R, percent);
            var intervalG = Lerp(fromRgba.G, toRgba.G, percent);
            var intervalB = Lerp(fromRgba.B, toRgba.B, percent);
            var intervalA = Lerp(fromRgba.A, toRgba.A, percent);
            return Color.FromRgba((byte)intervalR, (byte)intervalG, (byte)intervalB, (byte)intervalA);
        }

        private static Color[] tintColors;
        public void TintImage(Image image, float amount, float opacity)
        {
            if (amount == 0)
            {
                return;
            }
            if (tintColors == null)
            {
                var primaryColors = new[] {
                    Color.FromRgb(255, 0, 0),
                    Color.FromRgb(255, 255, 0),
                    Color.FromRgb(0, 255, 0),
                    Color.FromRgb(0, 255, 255),
                    Color.FromRgb(0, 0, 255),
                    Color.FromRgb(255, 0, 255),
                    Color.FromRgb(255, 0, 0)
                };
                tintColors = new Color[360];
                int index = 0;
                for (var x = 0; x < primaryColors.Length - 1; x++)
                {
                    var firstColor = primaryColors[x];
                    var secondColor = primaryColors[x + 1];
                    for (var y = 0; y < 60; y++)
                    {
                        tintColors[index++] = InterpolateColor(firstColor, secondColor, 60, y);
                    }
                }
            }
            var tintValue = (int)Math.Min(Math.Max((amount * 180) + 180, 0), 359);
            var tintColor = tintColors[tintValue].WithAlpha(opacity);
            image.Mutate(x => x.SetGraphicsOptions(o => o.ColorBlendingMode = PixelColorBlendingMode.Multiply).Fill(tintColor));
        }
```

```

**Comment by UkooLabs** [comment]
Now i just need to work out how do to the auto color / contrast as above

**Reply by C10B** [reply]
Did you ever sort out auto contrast or auto brightness?
